src/VsixTesting.Installer/Program.cs

```diff
@@ -649,4 +665,116 @@ internal static class AppDomainExtensions
         public static T CreateInstanceFromAndUnwrap<T>(this AppDomain domain)
             => (T)domain.CreateInstanceFromAndUnwrap(typeof(T).Assembly.Location, typeof(T).FullName);
     }
+
+    internal class VisualStudioUtil
+    {
+        public const string ProcessName = "devenv";
+
+        public static DTE GetDTE(Process process)
+        {
+            string namePattern = $@"!VisualStudio\.DTE\.(\d+\.\d+):{process.Id.ToString()}";
+            return (DTE)RunningObjectTable.GetRunningObjects(namePattern).FirstOrDefault();
+        }
+
+        public static IEnumerable<DTE> GetRunningDTEs()
+        {
+            foreach (var process in Process.GetProcessesByName(ProcessName))
+            {
+                if (GetDTE(process) is DTE dte)
+                    yield return dte;
+            }
+        }
+
+        public static DTE GetDteFromDebuggedProcess(Process process)
+        {
+            foreach (var dte in GetRunningDTEs())
```

Comment:
On the most recent release of Visual Studio, I'm finding the debugger doesn't always (or possibly ever) attach. I wonder if this is because `COMException` is being thrown and it's just continuing?

Rather than enumerating through the debuggers of all running instances, we could use `ProcessUtil.TryGetParentProcess` and retrieve the `DTE` directly using `GetDTE`.